### PR TITLE
Homework 6

### PR DESCRIPTION
**Pull request by BMartynas**
Added login page.
Implemented login functionality from front-end side.
Created 404 page and empty main page and set up routing for them.
Implemented authentication guard and protected main page with it.

### CODE REVIEW

File: frontend/src/app/auth/auth-form/auth-form.component.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        color: red;
+    }
+}
+
+::ng-deep .mat-form-field-appearance-outline .mat-form-field-outline {
+    background-color: white;
+    height: 3.43rem;
+    font-size: 1.125rem;
+    line-height: 1.3rem;
+    color: #474747;
```
Comment (AnastasiaHarost) [inline frontend/src/app/auth/auth-form/auth-form.component.scss:64]: add all colors to file with color-constants and use constants

File: frontend/src/app/auth/auth-form/auth-form.component.scss [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+:host ::ng-deep .mat-form-field-wrapper{
+    margin: 2.125rem 0 0 0 !important;
```
Comment (AnastasiaHarost) [inline frontend/src/app/auth/auth-form/auth-form.component.scss:68]: try to do it without important, it is a bad practice

File: frontend/src/app/auth/auth-form/auth-form.component.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<div class="container">
+    <div class="container__wrapper">
+            <h1>Budgetify</h1>
+            <form [formGroup]="loginForm" (ngSubmit)="onSubmit()" class="auth-form">
+                <mat-form-field class="auth-form__row" appearance="outline" hideRequiredMarker>
+                    <mat-label>Email</mat-label>
+                    <input matInput type="email" formControlName="email">
+                </mat-form-field>
+                <span *ngIf="loginForm.controls['email'].errors?.['required'] && loginForm.controls['email'].touched" class="auth-form__error-message">
```
Comment (AnastasiaHarost) [inline frontend/src/app/auth/auth-form/auth-form.component.html:9]: i think this check loginForm.controls['email'].touched is not needed
you can remove in all places

Comment (BMartynas) [reply frontend/src/app/auth/auth-form/auth-form.component.html:9]: Without it error message is displayed from the beginning when login page is opened, which is not a desired behavior.

**Pull request merged**